compendium/DeclarativeServices/src/manager/ConfigurationManager.cpp

```diff
@@ -73,6 +74,7 @@ namespace cppmicroservices
                         auto config = configAdmin->ListConfigurations("(pid=" + pid + ")");
                         if (config.size() > 0)
                         {
+                            changeCount = config.front()->GetChangeCount();
```

Comment:
is it possible that this `changeCount` variable can be overwritten with a value that still causes the race to occur. Since this code is looping over all the configurations associated with the declared pids, it seems possible that a hypothetical config1 has a change count of 2 and config2 has a change count of 1, causing the `ConfigurationManager::changeCount` to be 1. This appears like it could either cause the same race to occur (i.e. Modified is called when it shouldn't be) or it could cause Modified **not** to be called when it should.

Reply:
Yes I would agree on first glance. The issue boiled down is that the Configuration Manager manages multiple pids and we have stored just one. So we need to store a set of changeCounts, one for each pid. @jeffdiclemente

Reply:
@jeffdiclemente Two new tests added. Both fail prior to updating ChangeCount structure to map<string, unsigned long> rather than unsigned long. Both pass 10000 runs after update.